six/modules/c++/six.sidd/tests/test_byte_swap.cpp

```diff
@@ -0,0 +1,195 @@
+    parent->information->resolution.az = 0;
+    parent->information->collectionDuration = 0;
+    parent->information->collectionDateTime = six::DateTime();
+    parent->information->radarMode = six::RadarModeType::SPOTLIGHT;
+    parent->information->sensorName.clear();
+    parent->geometry.reset(new six::sidd::Geometry());
+    //parent->geometry->dopplerConeAngle = 45.8;
+
+    return derivedData;
+}
+
+void write(six::UByte* data, bool useStream, bool byteSwap)
+{
+    six::Container container(six::DataType::DERIVED);
+    container.addData(createData());
+
+    six::NITFWriteControl writer;
+    writer.getOptions().setParameter(six::WriteControl::OPT_BYTE_SWAP, (int)byteSwap);
+    writer.initialize(&container);
+
+    if (useStream)
+    {
+        io::ByteStream stream;
+        stream.write((sys::byte*)data, DATA_SIZE_IN_BYTES);
+        stream.seek(0, io::Seekable::START);
+        std::vector<io::InputStream*> streams;
+        streams.push_back(&stream);
+        writer.save(streams, OUTPUT_NAME);
+    }
+    else
+    {
+        writer.save(data, OUTPUT_NAME);
+    }
+}
+
+void read(const std::string& filename, six::UByte* data)
+{
+
+    six::NITFReadControl reader;
+    reader.load(filename);
+    six::Container* const container = reader.getContainer();
+    six::Data* const inData = container->getData(0);
+
+    const size_t numRows = inData->getNumRows();
+    const size_t numCols = inData->getNumCols();
+
+    six::Region region;
+    region.setStartRow(0);
+    region.setStartCol(0);
+    region.setNumRows(numRows);
+    region.setNumCols(numCols);
+    six::UByte buffer[DATA_SIZE_IN_BYTES];
+    region.setBuffer(buffer);
+    six::UByte* result = reader.interleaved(region, 0);
+
+    for (size_t ii = 0; ii < DATA_SIZE_IN_BYTES; ++ii)
+    {
+        data[ii] = result[ii];
+    }
+}
+
+bool run(bool useStream = false, bool byteswap = false)
+{
+    std::complex<float> complexData[DATA_LENGTH];
+    generateData(complexData);
+
+    six::UByte* sourceData = reinterpret_cast<six::UByte*>(&complexData[0]);
+    six::UByte testData[DATA_SIZE_IN_BYTES];
+    memcpy(testData, sourceData, DATA_SIZE_IN_BYTES);
+
+    if (!sys::isBigEndianSystem() && !byteswap || sys::isBigEndianSystem() && byteswap)
+    {
+        sys::byteSwap(sourceData, BYTES_PER_PIXEL, DATA_SIZE_IN_BYTES / BYTES_PER_PIXEL);
+    }
+
+    write(testData, useStream, byteswap);
+    read(OUTPUT_NAME, testData);
+
+    for (size_t ii = 0; ii < DATA_SIZE_IN_BYTES; ++ii)
+    {
+        if (sourceData[ii] != testData[ii])
+        {
+            return false;
+        }
+    }
+    return true;
+}
+}
+
+int main(int argc, char** argv)
+{
+    six::XMLControlFactory::getInstance().addCreator(
+        six::DataType::DERIVED,
+        new six::XMLControlCreatorT<
+        six::sidd::DerivedXMLControl>());
+
+    try
+    {
+        bool success = run(false, false) && run(true, false) &&
+            run(false, true) && run(false, false);
+
+        if (success) 
```

Comment:
Shorten this to just `return (success ? 0 : 1))` and get rid of the `return 1` at the end